### ISSUE TITLE
Issues with PythonQt 3.1

### ISSUE THREAD

**Issue by AlfredoRamos** [olav-st/screencloud-plugins#26]
I just upgrade to PythonQt 3.1 and now I get some errors with this library.

```
Error in file: /home/alfredo/.local/share/screencloud/ScreenCloud/plugins/file/main.py
Traceback (most recent call last):
  File "<string>", line 4, in <module>
ImportError: No module named 'PythonQt.QtUiTools'; 'PythonQt' is not a package
```
```
Error in file: /home/alfredo/.local/share/screencloud/ScreenCloud/plugins/file/main.py
Traceback (most recent call last):
  File "<string>", line 4, in <module>
ImportError: No module named 'PythonQt.QtUiTools'; 'PythonQt' is not a package
```

Running `import sys, Screencloud` from the [debug console](https://github.com/olav-st/screencloud/wiki/Plugin-Overview#the-debug-console):

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named 'Screencloud'
```

My system:
- Arch Linux x64
- GCC 6.1.1
- Qt 5.6.1
- Python 2.7.11 and 3.5.1

**Comment by olav-st** [comment]
> Running import sys, Screencloud from the debug console:

That should be 'ScreenCloud`, with a capital C. Try copy-pasting the commands from the wiki article.

**Comment by AlfredoRamos** [comment]
I am apparently blind...

```python
py> import sys, ScreenCloud
py> print(sys.version_info)
sys.version_info(major=3, minor=5, micro=1, releaselevel='final', serial=0)
py> print("\n".join(sys.modules))
os.path
importlib.util
PythonQt.QtGui
site
_sysconfigdata
sys
_string
_collections_abc
weakref
_weakref
struct
encodings.latin_1
_io
encodings.aliases
_struct
token
genericpath
_functools
PythonQt.QtXml
tokenize
_imp
PythonQt.Qt
__main__
os
reprlib
_collections
heapq
codecs
imp
builtins
zipimport
_weakrefset
stat
PythonQt.QtOpenGL
marshal
warnings
math
_sitebuiltins
copyreg
sre_parse
PythonQt.QtCore
PythonQt.QtXmlPatterns
PythonQt.QtNetwork
_thread
PythonQt.QtQuick
operator
PythonQt.QtWebKit
keyword
importlib.abc
importlib
_signal
errno
_hashlib
types
hashlib
_codecs
_random
sysconfig
PythonQt.private
_bootlocale
_locale
posix
encodings.utf_8
PythonQt.QtQml
_frozen_importlib_external
collections
posixpath
importlib._bootstrap
base64
string
importlib.machinery
sre_compile
random
encodings
_heapq
time
abc
itertools
_sre
_frozen_importlib
ScreenCloud
_operator
_warnings
re
functools
sre_constants
PythonQt.QtSvg
contextlib
_stat
binascii
collections.abc
importlib._bootstrap_external
PythonQt.QtMultimedia
io
PythonQt
file_uploader
PythonQt.QtSql
py> print("\n".join(sys.path))
/home/alfredo/.local/share/screencloud/ScreenCloud/plugins/file/modules
/usr/bin/modules
/usr/lib/python35.zip
/usr/lib/python3.5
/usr/lib/python3.5/plat-linux
/usr/lib/python3.5/lib-dynload
/usr/lib/python3.5/site-packages
py> print(ScreenCloud.getPluginDir())
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.5/site-packages/ScreenCloud.py", line 42, in getPluginDir
    return QDesktopServices.storageLocation(QDesktopServices.DataLocation) + "/plugins"
AttributeError: QDesktopServices has no attribute named 'storageLocation'
py>
```

In the [QtDesktopServices](https://doc.qt.io/qt-5/qdesktopservices.html) docs:

> Note: Since Qt 5, storageLocation() and displayName() are replaced by functionality provided by the QStandardPaths class.

I also maintain the [pythonqt-qt5](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=pythonqt-qt5) package for ScreenCloud, maybe I'm missing something, but I haven't found the "right" way to install this library in particular.

**Comment by olav-st** [comment]
> In the QtDesktopServices docs:
>  > Note: Since Qt 5, storageLocation() and displayName() are replaced by functionality provided by the QStandardPaths class.


I've fixed this with https://github.com/olav-st/screencloud/commit/ac4dbf309d898a667e22450db45f72952e9e4fed

**Comment by AlfredoRamos** [comment]
The `QDesktopServices`/`QStandardPaths` issue was fixed, however after upgrading Qt to 5.7 and recompile PythonQt and ScreenCloud, I get a segfault:

```
$ gdb -ex run screencloud
GNU gdb (GDB) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from screencloud...(no debugging symbols found)...done.
Starting program: /usr/bin/screencloud 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fffdeebb700 (LWP 9005)]
[New Thread 0x7fffd735c700 (LWP 9006)]
[New Thread 0x7fffd66b6700 (LWP 9007)]
[New Thread 0x7fffd560d700 (LWP 9008)]

Thread 1 "screencloud" received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) backtrace full
#0  0x0000000000000000 in ?? ()
No symbol table info available.
#1  0x00007ffff521340e in ?? () from /usr/lib/libQt5Widgets.so.5
No symbol table info available.
#2  0x00007ffff5213011 in ?? () from /usr/lib/libQt5Widgets.so.5
No symbol table info available.
#3  0x00007ffff53fc7a5 in ?? () from /usr/lib/libQt5Widgets.so.5
No symbol table info available.
#4  0x00007ffff53fc833 in ?? () from /usr/lib/libQt5Widgets.so.5
No symbol table info available.
#5  0x00007ffff53fc882 in ?? () from /usr/lib/libQt5Widgets.so.5
No symbol table info available.
#6  0x00000000004a80c7 in SystemTrayIcon::SystemTrayIcon(QObject*, QString) ()
No symbol table info available.
#7  0x0000000000449a49 in main ()
No symbol table info available.
(gdb) 
```

Now I'll have to wait until the PythonQt team add support for Qt 5.7

**Comment by AlfredoRamos** [comment]
It seems that it has been fixed upstream:

- https://sourceforge.net/p/pythonqt/code/438/
